src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+const exampleDinosaurData = require("../data/dinosaurs");
+const exampleTicketData = require("../data/tickets");
+const exampleRoomData = require("../data/rooms");
+
+/*
+Write a function that will organize all of the dinosaurs by their period key. 
+The function should return an object where the keys are the period values, and 
+the values are an array of IDs.
+*/
+
+/* Expected Result
+  {
+    "Late Jurassic": [
+      "YLtkN9R37",
+      "GGvO1X9Zeh",
+      "BFjjLjea-O",
+      "iOVNUcv-ww",
+      "V53DvdhV2A",
+    ],
+    "Late Cretaceous": [
+      "WHQcpcOj0G",
+      "GKl035EYKN",
+      "2GglUqKT0G",
+      "Pr6kc4Q_Xf",
+      "ft5Gs5izdq",
+      "wuL4ddBinQ",
+    ],
+    // ...
+  }
+*/
+
+function organizeDinosaursByPeriod(dinos, key) {
```

Comment:
It looks like you've added some kind of requirement for including an optional `key` in here. Neat!